iotedgedev/deploymentmanifest.py

```diff
@@ -216,37 +216,11 @@ def _validate_createOptions(self):
             current_module_validation_success = True
             try:
                 self.output.info("Validating createOptions for module %s" % module_name)
-                create_options = []
                 if "settings" in module_info and "createOptions" in module_info["settings"]:
-                    create_options_value = module_info["settings"]["createOptions"]
-                    if len(str(create_options_value)) > TWIN_VALUE_MAX_SIZE:
-                        current_module_validation_success = False
-                        self.output.info("Length of createOptions in module %s exceeds %d" % (module_name, TWIN_VALUE_MAX_SIZE))
-                    create_options.append(str(create_options_value))
-                    # Merge additional create options
-                    for i in range(1, TWIN_VALUE_MAX_CHUNKS):
-                        property_name = "createOptions0%d" % i
-                        if property_name in module_info["settings"]:
-                            create_options_value = module_info["settings"][property_name]
-                            if len(str(create_options_value)) > TWIN_VALUE_MAX_SIZE:
-                                current_module_validation_success = False
-                                self.output.info("Length of %s in module %s exceeds %d" % (property_name, module_name, TWIN_VALUE_MAX_SIZE))
-                            create_options.append(str(create_options_value))
-                        else:
-                            break
-                    # Verify createOptions is valid json
-                    create_options_string = "".join(create_options).strip()
-                    if not create_options_string.startswith('{'):
-                        current_module_validation_success = False
-                        self.output.info("createOptions of module %s should be an object" % module_name)
-                    else:
-                        try:
-                            json.loads(create_options_string)
-                            if current_module_validation_success:
-                                self.output.info("createOptions of module %s validation passed" % module_name)
-                        except ValueError as err:
-                            current_module_validation_success = False
-                            self.output.info("createOptions of module %s is not a valid JSON string. Error: %s" % (module_name, err))
+                    current_module_validation_success &= self._validate_createOptions_lengh(module_name, module_info)
```

Comment:
Consider creating a method called `_validate_create_options` that does both length and format validation.  Basically a helper around the two methods.

Reply:
Updated